Assets/Plugins/Android/Editor/PlatformSpecificConfigEditorAndroid.cs

```diff
@@ -28,34 +28,11 @@
 namespace PlayEveryWare.EpicOnlineServices
 {
 
-    public class PlatformSpecificConfigEditorAndroid : IPlatformSpecificConfigEditor
+    public class ConfigSectionAndroid : ConfigSection<EOSAndroidConfig>
     {
-        public static string ConfigFilename = "eos_android_config.json";
-        EOSConfigFile<EOSAndroidConfig> configFile;
+        public ConfigSectionAndroid() : base("Android", "eos_android_config.json") { }
 
-        [InitializeOnLoadMethod]
-        static void Register()
-        {
-            EpicOnlineServicesConfigEditor.AddPlatformSpecificConfigEditor(new PlatformSpecificConfigEditorAndroid());
-        }
```

Comment:
Instead of allowing Unity to control when different config editors are added to our editor window, we allow the `EditorWindow` instance to instantiate the ones it needs when it opens.

Reply:
Isn’t this Register for the editor windows under Edit->Preference->EOS Unity?
Do they still exist after the changes?

Reply:
That's actually a different thing - that gets done with a static method that gets the attribute `[SettingsProvider]` added to it. See [here](https://docs.unity3d.com/ScriptReference/SettingsProvider.html) for documentation on that. 

We actually do not consistently implement this across the different configuration windows that we have, and since we make no mention of accessing the options via `Edit` -> `Preference` -> `EOS Unity` in the documentation, I removed them until we can implement all of them.

But back to your original question, the register method is something that we engineered. We utilized the `[InitializeOnLoadMethod]` attribute like this. When Unity launches, it scans all scripts for the `[InitializeOnLoadMethod]` attribute and executes the static method that it's applied to. We leveraged this behavior combined with our static approach to make every "Platform Specific" configuration class register itself with the central configuration window `EpicOnlineServicesConfigEditor`. 

To fix issues that this PR addresses, removing the static-ness was a central goal. So now, instead of the platform specific configuration editors adding themselves to the main config window statically and only when Unity launches, the main config window instantiates each section when the window is actually opened.